lib/Interpreter/ValueExtractionSynthesizer.cpp

```diff
@@ -395,34 +408,54 @@ namespace {
     return Call.get();
   }
 
-  void ValueExtractionSynthesizer::FindAndCacheRuntimeDecls() {
+  static bool VSError(clang::Sema* Sema, clang::Expr* E, llvm::StringRef Err) {
+    DiagnosticsEngine& Diags = Sema->getDiagnostics();
+    Diags.Report(E->getLocStart(),
+                 Diags.getCustomDiagID(
+                     clang::DiagnosticsEngine::Level::Error,
+                     "ValueExtractionSynthesizer could not find: '%0'."))
+        << Err;
+    return false;
+  }
+
+  bool ValueExtractionSynthesizer::FindAndCacheRuntimeDecls(clang::Expr* E) {
     assert(!m_gClingVD && "Called multiple times!?");
     DeclContext* NSD = m_Context->getTranslationUnitDecl();
+    clang::VarDecl* clingVD = nullptr;
     if (m_Sema->getLangOpts().CPlusPlus) {
-      NSD = utils::Lookup::Namespace(m_Sema, "cling");
-      NSD = utils::Lookup::Namespace(m_Sema, "runtime", NSD);
-      m_gClingVD = cast<VarDecl>(utils::Lookup::Named(m_Sema, "gCling", NSD));
-      NSD = utils::Lookup::Namespace(m_Sema, "internal",NSD);
+      if (!(NSD = utils::Lookup::Namespace(m_Sema, "cling")))
+        return VSError(m_Sema, E, "cling namespace");
+      if (!(NSD = utils::Lookup::Namespace(m_Sema, "runtime", NSD)))
+        return VSError(m_Sema, E, "cling::runtime namespace");
+      if (!(clingVD = dyn_cast_or_null<VarDecl>(
+                utils::Lookup::Named(m_Sema, "gCling", NSD))))
+        return VSError(m_Sema, E, "cling::runtime::gCling");
+      if (!(NSD = utils::Lookup::Namespace(m_Sema, "internal", NSD)))
+        return VSError(m_Sema, E, "cling::runtime::internal namespace");
     }
     LookupResult R(*m_Sema, &m_Context->Idents.get("setValueNoAlloc"),
                    SourceLocation(), Sema::LookupOrdinaryName,
                    Sema::ForRedeclaration);
 
     m_Sema->LookupQualifiedName(R, NSD);
-    assert(!R.empty()
-           && "Cannot find cling::runtime::internal::setValueNoAlloc");
+    if (R.empty())
+      return VSError(m_Sema, E, "cling::runtime::internal::setValueNoAlloc");
```

Comment:
Do we really need an extra diagnostic here? Doesn't `LookupQualifiedName()` complain already?